libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -87,7 +87,30 @@ public struct RestResponse {
     }
 }
 
+extension RestRequest {
+  
+  /// Calculated property to determine if this request is a data retrieval request with a SOQL query.
+  /// All such queries will return a JSON decodable QueryResponseWrapper.
+  /// Implied contract is that all requests matching both properties here will be decodable via QueryResponseWrapper<Record>
+  public var isQueryRequest: Bool {
+    get {
+      return self.method == .GET && self.path.lowercased().hasSuffix("query")
```

Comment:
@codefriar Isn't search supported?

Reply:
Search should work as well. Hand't thought about it.